options/posix/generic/unistd-stubs.cpp

```diff
@@ -677,49 +677,65 @@ void sync(void) {
 	}
 }
 
-unsigned long sysconf(int number) {
+long sysconf(int number) {
+	if(mlibc::sys_sysconf) {
+		long ret = 0;
+
+		int e = mlibc::sys_sysconf(number, &ret);
+
+		if(e && e != EINVAL) {
+			errno = e;
+			return -1;
+		}
+
+		if(e != EINVAL) {
+			return ret;
+		}
+	}
+
+	/* default return values, if not overriden by sysdep */
 	switch(number) {
 		case _SC_ARG_MAX:
 			// On linux, it is defined to 2097152 in most cases, so define it to be 2097152
 			return 2097152;
 		case _SC_PAGE_SIZE:
 			return mlibc::page_size;
 		case _SC_OPEN_MAX:
-			struct rlimit ru;
-			if (getrlimit(RLIMIT_NOFILE, &ru) < 0) {
-				mlibc::infoLogger() << "\e[31mmlibc: sysconf(_SC_OPEN_MAX) returns arbitrary value 256\e[39m" << frg::endlog;
-				return 256;
-			} else {
-				return (ru.rlim_cur == RLIM_INFINITY) ? -1 : ru.rlim_cur;
-			}
+			mlibc::infoLogger() << "\e[31mmlibc: sysconf(_SC_OPEN_MAX) returns fallback value 256\e[39m" << frg::endlog;
+			return 256;
 		case _SC_PHYS_PAGES:
-			// TODO: actually return a proper value for _SC_PHYS_PAGES
-			mlibc::infoLogger() << "\e[31mmlibc: sysconf(_SC_PHYS_PAGES) returns arbitrary value 1024\e[39m" << frg::endlog;
+			mlibc::infoLogger() << "\e[31mmlibc: sysconf(_SC_PHYS_PAGES) returns fallback value 1024\e[39m" << frg::endlog;
 			return 1024;
 		case _SC_NPROCESSORS_ONLN:
-			// TODO: actually return a proper value for _SC_NPROCESSORS_ONLN
-			mlibc::infoLogger() << "\e[31mmlibc: sysconf(_SC_NPROCESSORS_ONLN) unconditionally returns 1\e[39m" << frg::endlog;
+			mlibc::infoLogger() << "\e[31mmlibc: sysconf(_SC_NPROCESSORS_ONLN) returns fallback value 1\e[39m" << frg::endlog;
```

Comment:
Realistically, this (and others) probably need some kind of sysdep callback; although _SC_NPROCESSORS_ONLN could probably be fetched by get_nprocs(), if we implement that function.

Reply:
I mean that's already added in this PR. The warning messages are a good enough solution.